platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/files/hashmap/HalfDiskHashMap.java

```diff
@@ -425,93 +456,198 @@ public DataFileReader endWriting() throws IOException {
                 storeName,
                 size,
                 oneTransactionsData.stream().mapToLong(BucketMutation::size).sum());
-
-        final ExecutorService flushExecutor = getFlushExecutor();
         final DataFileReader dataFileReader;
         if (size > 0) {
-            final Queue<ReadBucketResult> queue = new ConcurrentLinkedQueue<>();
-            final Iterator<IntObjectPair<BucketMutation>> iterator =
-                    oneTransactionsData.keyValuesView().iterator();
-
-            // read and update all buckets in parallel, write sequentially in random order
-            fileCollection.startWriting();
-            int processed = 0;
-            int inFlight = 0;
-            while (processed < size) {
-                // submit read tasks
-                while (inFlight < MAX_IN_FLIGHT && iterator.hasNext()) {
```

Comment:
`MAX_IN_FLIGHT` constant is no longer in use.

Reply:
Indeed, it is not. However, it will be used again in the next version of the fix

Reply:
The latest version of the fix respects `MAX_IN_FLIGHT` as the maximum number of buckets to be processed in parallel to minimize resource usage (every bucket has a heap byte buffer)